plugins/entando-plugin-jpversioning/src/main/java/org/entando/entando/plugins/jpversioning/services/contentsversioning/ContentVersioningService.java

```diff
@@ -36,21 +38,48 @@
     @Autowired
     private VersioningManager versioningManager;
 
+    @Autowired
+    private ContentService contentService;
+
     public PagedMetadata<ContentVersionDTO> getListContentVersions(String contentId, RestListRequest requestList) {
         logger.debug("LIST listContentVersions for content {} with req {}", contentId, requestList);
-        List<ContentVersionDTO> contentVersions = new ArrayList<>();
+        List<ContentVersionDTO> contentVersionDTOs = new ArrayList<>();
+        List<Long> contentVersions = new ArrayList<>();
         try {
-            contentVersions = versioningManager.getVersions(contentId).stream().map(v -> mapContentVersionToDTO(getContentVersion(v)))
+            contentVersions = versioningManager.getVersions(contentId);
+            contentVersionDTOs = requestList.getSublist(contentVersions).stream()
+                    .map(cv -> mapContentVersionToDTO(getContentVersion(cv)))
                     .collect(Collectors.toList());
         } catch (ApsSystemException e) {
             e.printStackTrace();
         }
-        final List<ContentVersionDTO> sublist = requestList.getSublist(contentVersions);
         PagedMetadata<ContentVersionDTO> pagedResults = new PagedMetadata<>(requestList, contentVersions.size());
-        pagedResults.setBody(sublist);
+        pagedResults.setBody(contentVersionDTOs);
         return pagedResults;
     }
 
+    public ContentVersion getContentVersion(Long versionId) {
+        try {
+            ContentVersion contentVersion = versioningManager.getVersion(versionId);
+            return contentVersion;
+        } catch (ApsSystemException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public ContentDto getContent(Long versionId) {
+        try {
+            final ContentVersion contentVersion = versioningManager.getVersion(versionId);
+            final Content content = versioningManager.getContent(contentVersion);
+            final ContentDto contentDto = contentService.getDtoBuilder().convert(content);
+            return contentDto;
+        } catch (ApsSystemException e) {
+            e.printStackTrace();
```

Comment:
Can you substitute this console output with a right log?

Reply:
Done